### PR TITLE
make click an optional dependency

### PR DESCRIPTION
**Pull request by tybug**
also adds a note about `pip install slider[cli]` to the page of the docs where the CLI is mentioned

### CODE REVIEW

File: slider/cli.py [hunk 1]
```diff
@@ -29,6 +27,10 @@ def maybe_show_progress(it, show_progress, **kwargs):
                 ...
     """
     if show_progress:
+        try:
+            import click
+        except ImportError:
+            raise ImportError("click must be installed to show a progressbar")
```
Comment (llllllllll) [inline slider/cli.py:33]: can you do a `raise from` on the original import error to no drop that context for users. There could be a case where the import error is for some weird reason that the user doesn't expect.

File: slider/__main__.py [hunk 2]
```diff
@@ -1,34 +1,34 @@
-import click
-
 from . import Library
 
+if __name__ == '__main__':
```
Comment (llllllllll) [inline slider/__main__.py:3]: Does this file get executed on a normal import, or only when used as an executable? I agree with the nicer import error, but I think we may be able to drop this branch.

File: slider/__main__.py [hunk 3]
```diff
@@ -1,34 +1,34 @@
...
+    try:
+        import click
+    except ImportError:
+        raise ImportError("click must be installed to use the slider cli")
```
Comment (llllllllll) [inline slider/__main__.py:7]: Like below, let's `raise from` the original error, but I like the additional context

**Review by llllllllll — COMMENTED** [review]
I think this change make a lot of sense, thanks for adding this.

File: slider/__main__.py (hunk 2 above)
Comment (tybug) [reply slider/__main__.py:3]: good catch, you're right.

**Pull request merged**